src/plugins/data/public/ui/dataset_selector/dataset_table.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiBasicTable, EuiFieldSearch, EuiLink, EuiText } from '@elastic/eui';
+import React, { useRef, useState } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { DataStructure } from '../../../common';
+import { getQueryService } from '../../services';
+import { DataStructureFetchOptions } from '../../query';
+
+interface DatasetTableProps {
+  path: DataStructure[];
+  setPath: (newPath: DataStructure[]) => void;
+  index: number;
+  explorerDataset: DataStructure | undefined;
+  selectDataStructure: (item: DataStructure | undefined, newPath: DataStructure[]) => Promise<void>;
+  fetchNextDataStructure: (
+    nextPath: DataStructure[],
+    dataType: string,
+    options?: DataStructureFetchOptions
+  ) => Promise<DataStructure>;
+}
+
+export const DatasetTable: React.FC<DatasetTableProps> = (props) => {
+  const datasetService = getQueryService().queryString.getDatasetService();
+  const [loading, setLoading] = useState(false);
+  const searchRef = useRef<HTMLInputElement | null>(null);
+
+  const initialSelectedIds = props.explorerDataset?.id.split(',');
+  const dataStructures = props.path[props.index].children || [];
+  const paginationToken = props.path[props.index].paginationToken;
+
+  const onTableChange = async (options: DataStructureFetchOptions) => {
+    const typeConfig = datasetService.getType(props.path[1].id);
+    if (!typeConfig) return;
+
+    setLoading(true);
+    await props
+      .fetchNextDataStructure(props.path, typeConfig.id, options)
+      .then((newDataStructure) => {
+        props.setPath([...props.path.slice(0, props.index), newDataStructure]);
+      })
+      .finally(() => setLoading(false));
+  };
+
+  return (
+    <div className="datasetTable">
+      <EuiFieldSearch
+        fullWidth
+        inputRef={(node) => (searchRef.current = node)}
+        onSearch={(value) => onTableChange({ search: value })}
+      />
+
+      <EuiBasicTable
+        items={dataStructures}
+        itemId="id"
+        columns={[{ field: 'title', name: 'Name' }]}
+        loading={loading}
+        isSelectable
+        selection={{
+          onSelectionChange: (items) => {
+            if (items.length === 0) {
+              props.selectDataStructure(undefined, props.path.slice(0, props.index + 1));
+              return;
+            }
+            if (!items.every((item) => item.type === items[0].type)) {
+              throw new Error('All items must be of the same type');
```

Comment:
yes!

should we confirm with ux the proper wording? since this is an important detail i feel like people will ask.

could we add a test for this one. i can see one day someone might try to add this feature and not ask why we didn't want this. the test can be the safeguard

Reply:
seems the EuiTable eats the error, it will probably need to be set some state and display error on UI. When we have some proper wording i can add a follow up PR. will create an issue for this

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8362